### PR TITLE
Various fixes

### LINKED ISSUE #208: Ballot: Should not be able to submit negative values
I'm able to assign a negative allocation to projects and submit my ballot that way. It should not be possible to have negative allocations. 
Screenshot 2024-09-25 at 12 09 56

### PR DESCRIPTION
**Pull request by vidvidvid**
- closes #208

### CODE REVIEW

**Comment by vidvidvid** [comment]
@geovgy i'd appreciate if you review the added [functionality](https://github.com/grants-stack-frontier/op-rf4-voting-ui/issues/180) on the ballot page & test it out. i added a new number input component. it'd be great to eventually split up this page into a ballot context & components before it evolves into its final form.
once you check if everything is fine, please merge it ✌️

File: src/components/ui/number-input.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      hideSpinButtons = false,
+      decimalPlaces,
+      value,
+      ...props
+    },
+    ref
+  ) => {
+    let formattedValue = value;
+
+    if (decimalPlaces !== undefined && value !== undefined && value !== '') {
```
Comment (geovgy) [inline src/components/ui/number-input.tsx]: Please remove this part. As mentioned [here](https://github.com/grants-stack-frontier/op-rf4-voting-ui/issues/181#issuecomment-2369897399), fixing the decimal places to 2 should be handled by the API, not the frontend. I am unable to edit the allocation to the amount I want and when I type numbers, I see the OP amount changing but the input amount does not.

**Review by geovgy — CHANGES_REQUESTED** [review]
I'm assuming once merged with the other PRs, all the other functionality in the ballot of all the reordering and such will work. I don't see all of them applied here in this branch.

**Comment by geovgy** [comment]
Also getting an error on redistributing the allocations moving a project to a new position in the ballot.

File: src/app/ballot/page.tsx [hunk 2]
```diff
@@ -233,8 +229,9 @@ function YourBallot() {
   const displayProjects = searchTerm ? filteredProjects : projectList;
 
   const isMovable =
-    distributionMethod !== DistributionMethod.IMPACT_GROUPS &&
-    distributionMethod !== DistributionMethod.CUSTOM;
+    distributionMethod === DistributionMethod.TOP_TO_BOTTOM ||
+    distributionMethod === DistributionMethod.TOP_WEIGHTED ||
+    distributionMethod === DistributionMethod.CUSTOM;
```
Comment (geovgy) [inline src/app/ballot/page.tsx]: This explains one of the issues I'm dealing with. This value was previously correct being:
```typescript=
const isMovable =
    distributionMethod !== DistributionMethod.IMPACT_GROUPS &&
    distributionMethod !== DistributionMethod.CUSTOM;
```

You should not be able to reorder the projects when `CUSTOM` distribution method is selected (issue #187).

File: src/components/ui/number-input.tsx (hunk 1 above)
Comment (vidvidvid) [reply src/components/ui/number-input.tsx]: removed. i'm still getting 10 decimal places from the backend though

**Comment by vidvidvid** [comment]
@geovgy please check again - now i only update the position when the # input changes. calling `redistribute()` afterwards always yields seemingly non-deterministic results that i can't understand and i have no clue how to make sense of it. i'd appreciate if you could wrap up this functionality.

**Pull request closed without merging**